### PR TITLE
Fixes for issue #6

### PR DESCRIPTION
**Pull request by linas**
Assorted fixes, per issue #6

### CODE REVIEW

**Comment by edyirdaw** [comment]
@linas , thanks for granting me write access. But since @mjsduncan doesnt contribute directly to the project, just for being cautious, you can reinstate him to his previous non-write access :D, unless you or he think otherwise.

I was planning to make a pull request with some modifications,  but since its now merged (although it can be reverted easily from the github UI),  I will commit some minor modifications soon. I will also list the feature additions/modifications made in the pull request below.

**Comment by linas** [comment]
OK.

The #1 most important fix would be to get rid of the json, and port to Atomese. If you don't understand why, let me know, I can explain. If you don't understand how, I can explain that too.

There are three ways to port to Atomese. 

* One is to open a connection to the cogserver, and just send whatever scheme you want, and parse the replies. This allows you to do pretty much anything at all. If you open the connection in "quiet" mode, then parsing the results will be easier, as you won't get the extra markup to make it human-readable. This requires you to write a brand-new parser, in js, for Atomese. This would be all-new code, no one has done this before.

* A second option is to do the above, but limit yourself to the `StorageNode` API.  This is almost the same as above, except that it's faster, tailored for good network performance and ... prints no error messages, making it harder to debug. See https://wiki.opencog.org/w/StorageNode half-way down the page.

* Either of the above would be written in js, and, because its generic, should probably be a brand-new js project, so that others can use it (and not just the explorer) I can create a git repo for you, to hold this new code.

* Write a brand-new cogserver plugin to return json. This would replace the RESTful thing.  The returned json would need to be designed to fix all the mistakes in the RESTful API.  This might be easier (??) than any of the above. The brand-new code would be in C++. I don't really recommend this option, but ... it wouldn't be that bad.  The API would be the same `StorageNode` API, except sending and receiving json instead of s-expressions.

So basically, if you want to write js, pick the first few options; if you want to write C++, pick the last option. I can help with any of these.
